src/newtask/Form.js

```diff
@@ -0,0 +1,65 @@
+import { useRef, useState } from "react";
+
+const Form = (props) => {
+  const [toggleButton, setToggleButton] = useState(true);
+  const [data, setData] = useState({
+    taskName: "",
+    dueDate: "",
+    priority: "low",
+  });
+
+  const buttonEl = useRef();
+
+  const submitForm = (e) => {
+    e.preventDefault();
+    props.getNewTaskHandler(data);
+    props.closeFormHandler();
+  };
+
+  const getData = (e) => {
+    setData({ ...data, [e.target.name]: e.target.value });
+  };
+
+  const validateLength = (e) => {
+    const textLength = e.target.value.length;
+    if (textLength < 1) {
+      setToggleButton(true);
+      alert("Your task needs to have a name.");
+    } else {
+      setToggleButton(false);
```

Comment:
it's normally considered bad practice to have a state that is based on another state.

you can directly use the text length in your jsx ([line 58](https://github.com/Joao-Matias/to-do-list/pull/2/files?diff=split&w=0#diff-e9d79a6926ffa4adb1a831c13a5db81c9c137a31ff9dd9f94474af953bab5295R58)):

```jsx
      <button ref={buttonEl} disabled={task.name.length === 0}>
```

for the example above, I've used the suggested naming convention in this PR.